packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -29,19 +33,57 @@ import ListViewOutline from './list-view-outline';
 
 export default function ListViewSidebar() {
 	const { setIsListViewOpened } = useDispatch( editPostStore );
+	const { clearSelectedBlock, selectBlock } = useDispatch( blockEditorStore );
+	const { hasBlockSelection, selectedBlockClientIds } = useSelect(
+		( select ) => ( {
+			hasBlockSelection:
+				!! select( blockEditorStore ).getBlockSelectionStart(),
+			selectedBlockClientIds:
+				select( blockEditorStore ).getSelectedBlockClientIds(),
+		} ),
+		[]
+	);
 
 	const focusOnMountRef = useFocusOnMount( 'firstElement' );
 	const headerFocusReturnRef = useFocusReturn();
 	const contentFocusReturnRef = useFocusReturn();
 
+	const [ tab, setTab ] = useState( 'list-view' );
+	const [ lastSelectedBlock, setLastSelectedBlock ] = useState();
+
+	useEffect( () => {
+		if ( selectedBlockClientIds?.length === 1 ) {
+			setLastSelectedBlock( selectedBlockClientIds[ 0 ] );
+		}
+	}, [] );
+
 	function closeOnEscape( event ) {
 		if ( event.keyCode === ESCAPE && ! event.defaultPrevented ) {
 			event.preventDefault();
 			setIsListViewOpened( false );
+
+			// If there is no longer a block selection, but a single block was selected
+			// before opening the list view, then select that block again.
+			if ( ! hasBlockSelection && lastSelectedBlock ) {
+				selectBlock( lastSelectedBlock );
+			}
 		}
 	}
 
-	const [ tab, setTab ] = useState( 'list-view' );
+	function clearSelectionOnEscape( event ) {
```

Comment:
What do you think if we try implementing this logic inside the shared `PrivateListView` component in `block-editor`? An immediate benefit is that we don't have to implement this for every editor we have 😆 . It might be possible to do this inside `use-block-selection`, if that makes sense.

Reply:
Yes to what @kevin940726 said :)

Reply:
Oh, good idea! That'd be much neater if we can consolidate it as part of the list view component. The only thing is that we'd probably still need to have the secondary sidebar component be aware of when it shouldn't close the sidebar based on an Escape key press, which means the logic might be sort of split a little across the components. If we don't mind that, then I think it's a better option 👍

Reply:
Could we do `event.stopPropagation()` and/or `event.preventDefault()` to solve that?

Reply:
Good thinking 👍 It might depend on which way the propagation goes, so I think I might have had the wrong idea in mind that I needed to handle things in the sidebar. I'll have a play around with it next week and will report back how it goes!